invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/PatronDetails/components/ItemsSearch/ItemsSearch.js

```diff
@@ -37,29 +38,31 @@ export default class ItemsSearch extends Component {
   };
 
   executeSearch = queryString => {
-    if (queryString) {
-      return this.fetchItems(queryString);
-    } else {
-      return this.fetchItems(this.props.queryString);
-    }
+    queryString = queryString || this.props.queryString;
+    this.setState({ prevSearchQuery: queryString });
+
+    return this.fetchItems(queryString);
   };
 
-  _onPasteHandler = e => {
+  _onPasteHandler = async e => {
```

Comment:
we should discuss if we use `Promise` or `await` - `async` or at least make sure to define cases in which we use one or another. 
BTW in this case what will happen if the `executeSearch` is not fulfilled?

Reply:
Yes, I agree. Should we keep using promises until we have discussed this?

If `executeSearch` would fail it would not be handled and I think an exception will be raised. For the cases where we need to catch errors it might be cleaner to use the `Promise` directly instead of wrapping code in a `try..catch`. In this case it should be fine because `this.fetchItems` already catches errors.

Reply:
I vote for `await/async` 👍 :)

Reply:
Use a more useful name than `e` i.e. if it refers to an event, `event` is a good name for it :)